src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingChannelExistenceChecker.java

```diff
@@ -0,0 +1,293 @@
+
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+      String requestPayload = objectMapper.writeValueAsString(channelExistenceCheckerRequest);
+
+      HttpUriRequest request =
+          requestBuilder.generateStreamingIngestPostRequest(
+              requestPayload,
+              CHANNEL_STATUS_ENDPOINT,
+              DEBUG_PREFIX /* Message passed on to Ingest SDK */);
+
+      ChannelExistenceCheckerResponse channelExistenceCheckerResponse;
+      try (CloseableHttpResponse closeableHttpResponse = httpClient.execute(request)) {
+
+        if (closeableHttpResponse == null) {
+          LOGGER.warn(
+              "Null response obtained in {} for channel:{}", CHANNEL_STATUS_ENDPOINT, channelName);
+          throw new SFException(ErrorCode.CHANNEL_STATUS_FAILURE);
+        }
+
+        // Handle the exceptional status code
+        HttpResponse httpResponse =
+            handleExceptionalStatus(
+                closeableHttpResponse,
+                null,
+                ServiceResponseHandler.ApiName.STREAMING_CHANNEL_STATUS,
+                httpClient,
+                request,
+                requestBuilder);
+
+        // Grab the string version of the response entity
+        String blob = consumeAndReturnResponseEntityAsString(httpResponse.getEntity());
+
+        // Read out our blob into a pojo
+        channelExistenceCheckerResponse =
+            objectMapper.readValue(blob, ChannelExistenceCheckerResponse.class);
+      }
+
+      if (channelExistenceCheckerResponse.getStatusCode() == SUCCESS.getStatusCode()) {
+        assert channelExistenceCheckerResponse.getChannels().size() == 1;
+        if (channelExistenceCheckerResponse
+            .getChannels()
+            .get(0)
+            .getStatusCode()
+            .equals(ERR_CHANNEL_DOES_NOT_EXIST_OR_IS_NOT_AUTHORIZED.getStatusCode())) {
+          LOGGER.debug(String.format("%s Channel:%s doesn't exist", DEBUG_PREFIX, channelName));
+          return false;
+        } else if (channelExistenceCheckerResponse
+                .getChannels()
+                .get(0)
+                .getStatusCode()
+                .equals(ERR_CHANNEL_HAS_INVALID_CLIENT_SEQUENCER.getStatusCode())
+            || channelExistenceCheckerResponse
+                .getChannels()
+                .get(0)
+                .getStatusCode()
+                .equals(SUCCESS.getStatusCode())) {
+          // We check two status codes, one for invalid client sequencers and one for a valid client
+          // Sequencer
+          LOGGER.debug(String.format("%s Channel:%s does exist", DEBUG_PREFIX, channelName));
+          return true;
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.error(
+          "{} Error Getting a valid response for ChannelName:{} error:{}",
+          DEBUG_PREFIX,
+          channelName,
+          e.getMessage());
```

Comment:
any issue with returning here instead of thrown an error?

Reply:
I think that's ok, being unable to get a channel basically means it doesnt exist. We should add that behavior to the javadoc though: @return True if it exists, false if it doesn't or failed to get status